### PR TITLE
Fix numerical issues in TF and JAX `moments` implementation.

### PR DESCRIPTION
**Pull request by 3lim**

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. What was the motivation for this change, did you encounter an issue?

**Comment by 3lim** [comment]
This fixes the same underlying issue as in https://github.com/keras-team/tf-keras/commit/cdf256d4ac7e25bb64e736af6d12d4d73ea77978 for Keras 3.

**Comment by 3lim** [comment]
For context: Without this fix, it won't guarantee that variance >= 0 for example, which causes issues when used with BatchNorm.

**Comment by 3lim** [comment]
@fchollet Friendly ping on this. Without this, we're occasionally running into NaN's when using BatchNorm.

**Review by fchollet — APPROVED** [review]
LGTM, the changes are fine.
